lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -27,8 +27,8 @@ def start
 
       # pipe the output to tee while capturing the exit code in a file
       script = <<-SCRIPT
-          (#{su_prefix}#{remote_script} < /dev/null; echo $?>#{exit_code_path}) | /usr/bin/tee #{output_path}
```

Comment:
`sh` doesn't have `$status` (at least when I tried that). When trying csh, I've seen `$?` working. Which version were you testing with?

```
$ csh --version
tcsh 6.19.00 (Astron) 2015-05-21 (x86_64-unknown-linux) options wide,nls,dl,al,kan,sm,rh,color,filec
$ echo $?
0
```